main.cpp

```diff
@@ -42,16 +42,64 @@ struct Point
     }
 private:
     float x{0}, y{0};
-};
+};
+
+
 
 template<typename Type>
 struct Wrapper
 {
     Wrapper(Type&& t) : val(std::move(t)) 
     { 
         std::cout << "Wrapper(" << typeid(val).name() << ")" << std::endl; 
-    }
-};
+    }
+
+    void print()
+    {
+        std::cout << "Wrapper::print(" << val << ")" << std::endl;
+    }
+
+    Type val{}; 
+    
+};
+// 8) You will need to specialize the Wrapper class template to work with the Point class.
+//a separate class definition entirely
+#include <memory>
+#
+template<>
+struct Wrapper<Point>
+{
+    using Type = Point;
+    Wrapper(Type&& v) : v_(std::move(v))
+    {
+        std::cout << "Wrapper(" << typeid(v_).name() << ")" << std::endl; 
+    }
+
+    void print()
+    {
+        std::cout << "Wrapper::print(" << v_.toString() << ")" << std::endl;
+    }
+
+    Type v_{0.f, 0.f};
+
+};
```

Comment:
Project 9.1 Challenge 1:
There is a lot of repetition between the template class `Wrapper`, and the template specialization for `Wrapper<Point>`
Can you figure out how to eliminate the duplicate constructor?
hint: you only have to _explicitly specialize_ that which is unique.
You can specialize member functions without specializing the entire class :wink: